docs/rfcs/2023-09-29-merge-scenarios.md

```diff
@@ -0,0 +1,149 @@
+- Contact and group member records (possibly for many groups) not being merged when contact connects and group member records already exist.
+- Group member records in different groups not being merged if contact doesn't exist.
+- Duplicate contacts, or contact and group member records not being merged when connecting via contact address present in contact/group member profile. Probing could be avoided in this case as client already has knowledge of contact address "identity".
+
+This problem is a direct consequence of lack of user identity in the platform, and in some cases we even consider it a feature. For example, duplicate contacts via repeat connections can be used for having conversation scopes. Though in general it seems to bring more confusion. It also limits some interactions, such as sending direct message to group members, or viewing a list of groups contact is member of (not implemented).
+
+On the other hand, solving all these cases reduces the privacy of the main profile, since the client cooperates with other probing clients blindly. To keep this property, we can add an opt-out "Merge contacts" client setting which would affect existing and new probe mechanisms. (It would act as an Incognito mode currently does - launch probes w/t launching probe hashes, and never confirm received probes) We can also ignore it, since this can be worked around via Incognito profiles or multiple user profiles.
+
+There is one more problem that could be addressed in the same scope - repeat group join via group link fails if the contact with host wasn't deleted. This happens due to group links re-using contact address machinery together with prohibiting repeat connections. This could be treated in a similar way to how it is treated for contact addresses: instead of simply prohibiting repeat connection, if group exists, it would be opened; if group doesn't exist, client would send host a request to re-invite them.
+
+## Solution
+
+### Duplicate contacts on repeat connections via invitation links
+
+Can be solved by probing contacts.
+
+### Duplicate contacts on repeat connection via contact request
+
+Can be solved by probing contacts.
+
+Special case - records can be directly associated w/t probing when address is known in a contact / group member profile, see below.
+
+### Contact and group member records not merged when contact connects and group member records already exist
+
+Can be solved by probing group members.
+
+Send probe hashes to all viable group members? Or should group member records already have been merged between each other? (see below)
+
+In all cases above - who should initiate probing? It doesn't matter much, but possibly the contact that started connection.
+
+### Group member records in different groups not being merged if contact doesn't exist
```

Comment:
Don't merge group members at all - merge contacts to group members and contacts with each other?